Configure logrotate when systemd is not used

elemoine opened the pull request.

Commit:
Configure logrotate

This adds a logrotate configuration for Heka when systemd is not used.

elemoine commented:
I removed the `su` option as suggested by @SwannCroiset in https://github.com/tcpcloud/salt-formula-collectd/pull/39/files.

SwannCroiset commented on heka/files/heka_logrotate.conf:
```diff
@@ -0,0 +1,19 @@
+  # number of rotated files to keep
+  rotate 10
+  # do not rotate files unless both size and time conditions are met
+  hourly
```
'hourly' is useless b/c logrotate is executed daily basis.
If we absolutely need  to rotate hourly we must run a dedicated crontab for that (and don't put this conf in standard logrotate conf directory).
see https://github.com/openstack/fuel-plugin-lma-collector/blob/bb67a130628477de4e3d23e9490ed854e7fd959f/deployment_scripts/puppet/modules/heka/manifests/init.pp#L224-L252

and 
https://github.com/openstack/fuel-plugin-lma-collector/blob/bb67a130628477de4e3d23e9490ed854e7fd959f/deployment_scripts/puppet/modules/heka/templates/logrotate.cron.erb#L33

elemoine replied:
I guess we need to do the same as the Fuel plugin, or the same problems will eventually resurface. Thanks a lot for catching this.

Commit:
Add specific logrotate script and crontab

The log rate produced by Heka may be high when the Elasticsearch output
queue is full. For that reason, we want to rotate log files every hour.
By default, logrotate runs daily on Ubuntu. So we use a specific
logrotate script and crontab.

elemoine commented:
@SwannCroiset I added a commit that introduces the specific logrotate script and crontab, as done by the Fuel plugin.
